tests_and_analysis/test/euphonic_test/test_qpoint_frequencies.py

```diff
@@ -435,19 +441,53 @@ def test_calculate_dos(
              'lzo_222_full_mode_widths.json',
              'lzo_222_full_adaptive_dos.json',
              np.arange(0, 100, 0.1)*ureg('meV'))])
-    def test_calculate_dos_with_mode_widths(
+    def test_calculate_dos_with_mode_widths_ref_method(
             self, material, qpt_freqs_json, mode_widths_json,
             expected_dos_json, ebins):
         qpt_freqs = get_qpt_freqs(material, qpt_freqs_json)
-        with open(os.path.join(get_fc_dir(), mode_widths_json), 'r') as fp:
-            modw_dict = json.load(fp)
-        mode_widths = modw_dict['mode_widths']*ureg(
-            modw_dict['mode_widths_unit'])
+        mode_widths = get_mode_widths(mode_widths_json)
         dos = qpt_freqs.calculate_dos(
-            ebins, mode_widths=mode_widths)
+            ebins, mode_widths=mode_widths, adaptive_method='reference')
         expected_dos = get_expected_spectrum1d(expected_dos_json)
         check_spectrum1d(dos, expected_dos)
 
+    @pytest.mark.parametrize(
+        ('material, qpt_freqs_json, mode_widths_json, '
+         'expected_dos_json, ebins'), [
+            ('quartz', 'quartz_554_full_qpoint_frequencies.json',
+             'quartz_554_full_mode_widths.json',
+             'quartz_554_full_adaptive_dos_fast.json',
+             np.arange(0, 155, 0.1)*ureg('meV')),
+            ('LZO', 'lzo_222_full_qpoint_frequencies.json',
+             'lzo_222_full_mode_widths.json',
+             'lzo_222_full_adaptive_dos_fast.json',
+             np.arange(0, 100, 0.1)*ureg('meV'))])
+    def test_calculate_dos_with_mode_widths_fast_method(
+            self, material, qpt_freqs_json, mode_widths_json,
+            expected_dos_json, ebins):
+        qpt_freqs = get_qpt_freqs(material, qpt_freqs_json)
+        mode_widths = get_mode_widths(mode_widths_json)
+        dos = qpt_freqs.calculate_dos(
+            ebins, mode_widths=mode_widths, adaptive_method='fast')
+        expected_dos = get_expected_spectrum1d(expected_dos_json)
+        check_spectrum1d(dos, expected_dos, 1e-13)
```

Comment:
To avoid duplication, these parameters can probably be used in the existing `test_calculate_dos_with_mode_widths`. We just need another parameter to set the other kwargs to `calculate_dos`. Does that make sense?